defi/lending_platform/src/lending_platform.rs

```diff
@@ -0,0 +1,436 @@
+                };
+            self.users.insert(user_badge.resource_address(), user);
+            info!(
+                "[LendingPlatform][POOL] Created new user with an address of {}.",
+                user_badge.resource_address()
+            );
+            user_badge
+        }
+
+        #[auth(admin_badge)]
+        pub fn new_asset(&self, asset_address: Address, ltv_ratio: Decimal) {
+            assert! (
+                ltv_ratio >= 0.into() && ltv_ratio <= 1.into(),
+                "[LendingPlatform][POOL] LTV Ratio must be between 0.0 & 1.0 (inclusive)."
+                );
+            match self.asset_ltv_ratios.get(&asset_address) {
+                Some(..) =>  {
+                    panic!("[LendingPlatform][POOL] Asset {} already exists in lending pool.", asset_address);
+                },
+                None => {
+                    self.asset_ltv_ratios.insert(asset_address, ltv_ratio);
+                    info!(
+                        "[LendingPlatform][POOL] Added asset {} with LTV ratio of {} to the lending pool.",
+                        asset_address,
+                        ltv_ratio
+                    )
+                }
+            };
+        }
+
+        pub fn deposit_asset(&self, user_auth: BucketRef, asset: Bucket) -> BucketRef {
+            let asset_address = asset.resource_address();
+            let amount = asset.amount();
+
+            let mut user = self.users.get(&user_auth.resource_address()).unwrap();
+            let user_address = user.user_address;
+
+            match self.assets.get(&asset_address) {
+                Some(x) =>  x.put(asset),
+                None => {
+                    let vault = Vault::with_bucket(asset);
+                    self.assets.insert(asset_address, vault);
+                }
+            };
+
+            // Update User Deposit Balance
+            user.increase_deposit_balance(asset_address, amount);
+            self.users.insert(user_auth.resource_address(), user);
+            info!(
+                "[LendingPlatform][USER:{}] Depositing {} of {}.",
+                user_address,
+                amount,
+                asset_address
+            );
+
+            info!(
+                "[LendingPlatform][POOL] New liquidity balance for {} is {}.",
+                asset_address,
+                self.assets.get(&asset_address).unwrap().amount()
+            );
+            user_auth
+        }
+
+        pub fn withdrawal_asset(
+            &self,
+            user_auth: BucketRef,
+            asset_address: Address,
+            amount: Decimal
+        ) -> (BucketRef, Bucket) {
+            let mut user = self.users.get(&user_auth.resource_address()).unwrap();
+            let user_address = user.user_address;
+
+            let removed_asset = match self.assets.get(&asset_address) {
+                Some(x) =>  {
+                    assert! (
+                        x.amount() >= amount,
```

Comment:
Shouldn't you take the `asset_ltv_ratios` into account here to make sure the user is over-collateralizing their loan ?